### PR TITLE
[MM-38469] - fixed several console errors regarding propTypes

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
this PR fixes several console errors that came up due to a false setting of props within the `Menu.ItemToggleModalRedux` component

#### Ticket Link
[MM-38469](https://mattermost.atlassian.net/browse/MM-38469)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by jespino — APPROVED** [review]
LGTM

**Review by catalintomai — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @michelengelen 
Tested, looks good to merge.
- Verified listed console errors are no longer reproducing.

File: components/widgets/menu/menu_items/menu_item_toggle_modal_redux.tsx [hunk 1]
```diff
@@ -16,18 +16,18 @@ type Props = {
     dialogType: React.ComponentType<any>;
     dialogProps?: Dictionary<any>;
     extraText?: string;
-    text?: string;
-    accessibilityLabel?: string;
+    text?: React.ReactNode;
+    ariaLabel?: string;
```
Comment (saturninoabril) [inline components/widgets/menu/menu_items/menu_item_toggle_modal_redux.tsx:20]: @michelengelen This breaks accessibility in several places. Like in the screenshot, it only reflect ` dialog` where previously it was `Custom Status dialog`.

Screen Shot 2021-09-29 at 2 23 56 PM

**Pull request merged**